modules/gridBidAdapter.js

```diff
@@ -224,75 +129,403 @@ function _getBidFromResponse(respItem) {
+  let jwpseg = null;
+  let content = null;
+  let schain = null;
+  let userId = null;
+  let user = null;
+  let userExt = null;
+  let {bidderRequestId, auctionId, gdprConsent, uspConsent, timeout, refererInfo} = bidderRequest;
+
+  const referer = refererInfo ? encodeURIComponent(refererInfo.referer) : '';
+  const imp = [];
+  const bidsMap = {};
+
+  validBidRequests.forEach((bid) => {
+    if (!bidderRequestId) {
+      bidderRequestId = bid.bidderRequestId;
+    }
+    if (!auctionId) {
+      auctionId = bid.auctionId;
+    }
+    if (!schain) {
+      schain = bid.schain;
+    }
+    if (!userId) {
+      userId = bid.userId;
+    }
+    const {params: {uid, keywords}, mediaTypes, bidId, adUnitCode, realTimeData} = bid;
+    bidsMap[bidId] = bid;
+    if (!pageKeywords && !utils.isEmpty(keywords)) {
+      pageKeywords = utils.transformBidderParamKeywords(keywords);
+    }
+    if (realTimeData && realTimeData.jwTargeting) {
+      if (!jwpseg && realTimeData.jwTargeting.segments) {
+        jwpseg = realTimeData.segments;
+      }
+      if (!content && realTimeData.content) {
+        content = realTimeData.content;
+      }
+    }
+    let impObj = {
+      id: bidId,
+      tagid: uid.toString(),
+      ext: {
+        divid: adUnitCode
+      }
+    };
+
+    if (!mediaTypes || mediaTypes[BANNER]) {
+      const banner = createBannerRequest(bid, mediaTypes ? mediaTypes[BANNER] : {});
+      if (banner) {
+        impObj.banner = banner;
+      }
+    }
+    if (mediaTypes && mediaTypes[VIDEO]) {
+      const video = createVideoRequest(bid, mediaTypes[VIDEO]);
+      if (video) {
+        impObj.video = video;
+      }
+    }
+
+    if (impObj.banner || impObj.video) {
+      imp.push(impObj);
+    }
+  });
+
+  const source = {
+    tid: auctionId,
+    ext: {
+      wrapper: 'Prebid_js',
+      wrapper_version: '$prebid.version$'
+    }
+  };
+
+  if (schain) {
+    source.ext.schain = schain;
+  }
+
+  const tmax = config.getConfig('bidderTimeout') || timeout;
+
+  let request = {
+    id: bidderRequestId,
+    site: {
+      page: referer
+    },
+    tmax,
+    source,
+    imp
+  };
+
+  if (content) {
+    request.site.content = content;
+  }
+
+  if (jwpseg && jwpseg.length) {
+    user = {
+      data: [{
+        name: 'iow_labs_pub_data',
+        segment: jwpseg.map((seg) => {
+          return {name: 'jwpseg', value: seg};
+        })
+      }]
+    };
+  }
+
+  if (gdprConsent && gdprConsent.consentString) {
+    userExt = {consent: gdprConsent.consentString};
+  }
+
+  if (userId) {
+    userExt = userExt || {};
+    if (userId.tdid) {
+      userExt.unifiedid = userId.tdid;
+    }
+    if (userId.id5id) {
+      userExt.id5id = userId.id5id;
+    }
+    if (userId.digitrustid && userId.digitrustid.data && userId.digitrustid.data.id) {
+      userExt.digitrustid = userId.digitrustid.data.id;
+    }
+    if (userId.lipb && userId.lipb.lipbid) {
+      userExt.liveintentid = userId.lipb.lipbid;
+    }
```

Comment:
Update your docs, at the moment it says that your adapter doesn't support any user id, also please add some test case for user ids. At the moment your tests cases are not covering this branch.